processor/README.md

```diff
@@ -132,24 +132,29 @@ if the span should be included or excluded from the processor. By default, all
 spans are processed by the processor.
 
 To configure this option, under `include` and/or `exclude`:
-- at least one of or both `services` and `attributes` is required.
+- at least one of `services`, `span_names` or `attributes` is required.
 
 Note: If both `include` and `exclude` are specified, the `include` properties
 are checked before the `exclude` properties.
 
+Items in `span_names` array are glob patterns, with wildcard characters * and ?
```

Comment:
I understand that we have a desire to expedite this; however, we should add a field that says match and specifies how we do the matching. That way we can extend globing/exact to services and attributes which would probably want the same option.

Reply:
I am hesitant to introduce too many different ways to do matching. My preference is if possible to have one way that is flexible enough and fast enough. Regex can do everything that strict and glob matching can do, so it satisfies the flexibility criteria. The concern with regex-es is performance and backward compatibility. Performance differences between globs and regexps for equivalent match strings are small and I am inclined to ignore them. Compatibility-wise this is a minor breaking change and I think it is acceptable for Alpha product.

Reply:
Regardless of us supporting regex - we should have it be an explicit field stating how to do the matching. (glob or exact)

Reply:
Unclear why you think so. Please expand.

My opinion is that we need to have minimal number of config options that allow the user to cover their use cases. Large number of config options and code that implements them is a liability and I prefer to minimize it if possible.

Reply:
Because it is clear to the user how things are to be matched. We shouldn't expose globbing to only one field. It should be available to everything. I guarantee that there are use cases for globbing by service name or attributes. By making a special case for one field, we only open it up to requiring a further refactor down the line. This processor is clean because it is verbose and clear about its actions.

Reply:
Consistency is a good point and I was intending to make corresponding modifications elsewhere. Other filters will also use regexes.